tracer/src/Datadog.Trace/AppSec/Waf/Waf.cs

```diff
@@ -51,20 +50,69 @@ internal Waf(IntPtr wafHandle, WafLibraryInvoker wafLibraryInvoker)
         /// <param name="rulesFile">can be null, means use rules embedded in the manifest </param>
         /// <param name="rulesJson">can be null. RemoteConfig rules json. Takes precedence over rulesFile </param>
         /// <returns>the waf wrapper around waf native</returns>
-        internal static InitializationResult Create(WafLibraryInvoker wafLibraryInvoker, string obfuscationParameterKeyRegex, string obfuscationParameterValueRegex, string? rulesFile = null, string? rulesJson = null)
+        internal static InitOrUpdateResult Create(WafLibraryInvoker wafLibraryInvoker, string obfuscationParameterKeyRegex, string obfuscationParameterValueRegex, string? rulesFile = null, string? rulesJson = null)
         {
             var wafConfigurator = new WafConfigurator(wafLibraryInvoker);
-            InitializationResult initializationResult;
+            InitOrUpdateResult initOrUpdateResult;
             if (!string.IsNullOrEmpty(rulesJson))
             {
-                initializationResult = wafConfigurator.ConfigureFromRemoteConfig(rulesJson, obfuscationParameterKeyRegex, obfuscationParameterValueRegex);
+                initOrUpdateResult = wafConfigurator.ConfigureFromRemoteConfig(rulesJson, obfuscationParameterKeyRegex, obfuscationParameterValueRegex);
             }
             else
             {
-                initializationResult = wafConfigurator.Configure(rulesFile, obfuscationParameterKeyRegex, obfuscationParameterValueRegex);
+                initOrUpdateResult = wafConfigurator.Configure(rulesFile, obfuscationParameterKeyRegex, obfuscationParameterValueRegex);
             }
 
-            return initializationResult;
+            return initOrUpdateResult;
+        }
+
+        public InitOrUpdateResult UpdateRules(string rules)
+        {
+            var rulesetInfo = new DdwafRuleSetInfo();
+            try
+            {
+                var argCache = new List<Obj>();
+                using var rulesObj = _wafConfigurator.GetConfigObjFromRemoteJson(rules, argCache);
+                var wafHandle = _wafLibraryInvoker.Update(_wafHandle, rulesObj.RawPtr, rulesetInfo);
```

Comment:
Probably outside the scope of this PR, but I think ultimately we should aim to have one method that accepts all types of WAF update and call it once.

Reply:
Yes, it's just a first step to facilitate it  https://datadoghq.atlassian.net/browse/APPSEC-7904